build.gradle

```diff
@@ -36,4 +38,28 @@ task clean(type: Delete) {
     delete rootProject.buildDir
 }
 
+apiValidation {
+    ignoredPackages += [
+            'io/customer/base/extenstions',
+            'io/customer/sdk/api',
+            'io/customer/sdk/data/store',
+            'io/customer/sdk/queue',
+            'io/customer/sdk/extensions',
+            'io/customer/sdk/util',
+    ]
+
+    ignoredProjects += [
+            'common-test',
+            'app',
+    ]
+
+    /**
+     * Classes (fully qualified) that are excluded from public API dumps even if they
+     * contain public API.
+     */
+    ignoredClasses.add("io.customer.messagingpush.BuildConfig")
+    ignoredClasses.add("io.customer.messaginginapp.BuildConfig")
+    ignoredClasses.add("io.customer.sdk.BuildConfig")
```

Comment:
Along with my previous comment about `ignoredProjects` comments, a comment here would also be great. I think I know why these files should be ignored but I am not sure what other files should be besides BuildConfig if I ever need to do that in the future.